lightning/src/ln/channel.rs

```diff
@@ -4026,21 +4049,15 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 			ChannelSignerType::Ecdsa(ref ecdsa) => {
 				ecdsa.sign_counterparty_commitment(&counterparty_initial_commitment_tx, Vec::new(), Vec::new(), &self.secp_ctx)
 					.map(|(signature, _)| signature)
-					.map_err(|_| ChannelError::Close(
-						(
-							"Failed to get signatures for new commitment_signed".to_owned(),
-							ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
-						)))
+					.ok()
```

Comment:
What's the reasoning for discarding error info here? The signer may return an error.

Reply:
Signer returning an error indicates async signing. At least that's how the other async signing operates. Looks like the `EcdsaChannelSigner::sign_counterparty_commitment` docs weren't updated like `ChannelSigner::get_per_commitment_point` were. Let me do that in this PR.

@TheBlueMatt Wonder if we should change the interface to use an `Option` instead of a `Result`once `async_signing` configuration parameter is dropped?